parser.ts

```diff
@@ -232,38 +242,112 @@ export function traverseExprHelper(c: TreeCursor, s: string, env: ParserEnv): Ex
           op = UniOp.Not;
           break;
         default:
-          throw new Error("Could not parse op at " + c.from + " " + c.to + ": " + s.substring(c.from, c.to))
+          throw new Error(
+            "Could not parse op at " +
+              c.from +
+              " " +
+              c.to +
+              ": " +
+              s.substring(c.from, c.to)
+          );
       }
       c.nextSibling(); // go to expr
       var expr = traverseExpr(c, s, env);
       c.parent();
       return {
         tag: "uniop",
         op: op,
-        expr: expr
-      }
+        expr: expr,
+      };
     case "MemberExpression":
       c.firstChild(); // Focus on object
       var objExpr = traverseExpr(c, s, env);
-      c.nextSibling(); // Focus on .
-      c.nextSibling(); // Focus on property
-      var propName = s.substring(c.from, c.to);
-      c.parent();
-      return {
-        tag: "lookup",
-        obj: objExpr,
-        field: propName
+      c.nextSibling(); // Focus on . or [
+      if (s.substring(c.from, c.to) == "[") {
+        // Start with :
+        c.nextSibling(); // Focus on start index or : or index
+        if (s.substring(c.from, c.to) == ":") {
+          c.nextSibling(); // Focus on end index or ]
+          if (s.substring(c.from, c.to) == "]") {
+            c.parent();
+            return { tag: "slice", obj: objExpr };
+          }
+          var endIndex = traverseExpr(c, s, env);
+          c.parent();
+          return { tag: "slice", obj: objExpr, index_e: endIndex };
+        }
+
+        // Start index or index
+        var startIndex = traverseExpr(c, s, env);
+        c.nextSibling(); // Focus on : or ]
+        if (s.substring(c.from, c.to) == "]") {
+          c.parent();
+          return { tag: "index", obj: objExpr, index: startIndex };
```

Comment:
We are set/dict group. We'll probably use your "index" expr in AST for our dictionary add (Because it's also a "[]" expression and we don't want to make more conflicts).
However, it seems that you have a global var "env" in parser. I'm not sure how this will be merged. But I believe that our two groups will have overlapping when dealing with "[]".